tests/unit/plugins/test_publisher.py

```diff
@@ -16,21 +16,476 @@
+        "deletions": [
+            {
+                "kind": "dns#resourceRecordSet",
+                "name": "service.nurit.com.",
+                "type": "A",
+                "ttl": 3600,
+                "rrdatas": [
+                    "127.10.20.23"
+                ]
+            }
+        ],
+        "startTime": "2018-05-01T18:41:51.577Z",
+        "id": "13",
+        "status": "pending"
+    }"""
+
+
+@pytest.fixture
+def resp_watch_status_update_record():
+    return {'kind': 'dns#change',
+            'additions': [
+                {'kind': 'dns#resourceRecordSet',
+                 'name': 'service.nurit.com.',
+                 'type': 'A',
+                 'ttl': 3600,
+                 'rrdatas': ['127.10.20.22', '127.10.20.24']}],
+            'deletions': [
+                {'kind': 'dns#resourceRecordSet',
+                 'name': 'service.nurit.com.',
+                 'type': 'A',
+                 'ttl': 3600,
+                 'rrdatas': ['127.10.20.22', '127.10.20.23']}],
+            'startTime': '2018-05-01T18:41:51.577Z',
+            'id': '13',
+            'status': 'done'}
+
+
+@pytest.fixture
+def resp_get_json_resource_record_sets():
+    return {
+        'kind': 'dns#resourceRecordSetsListResponse',
+        'rrsets': [
+            {'kind': 'dns#resourceRecordSet',
+             'name': 'nurit.com.',
+             'type': 'SOA', 'ttl': 21600,
+             'rrdatas': ['ns-cloud-c1.googledomains.com.'
+                         ' cloud-dns-hostmaster.google.com. '
+                         '1 21600 3600 259200 300']},
+            {'kind': 'dns#resourceRecordSet',
+             'name': 'service.nurit.com.',
+             'type': 'A', 'ttl': 3600,
+             'rrdatas': ['127.10.20.22', '127.10.20.23']}]
+    }
+
+
+@pytest.fixture
+def pubsub_message(mocker):
+    pubsub_msg = mocker.MagicMock(pubsub_v1.subscriber.message.Message)
+    pubsub_msg.message_id = 1234
+    return pubsub_msg
+
+
+@pytest.fixture
+def event_message(mocker, event_msg_data, pubsub_message):
+    event_msg = mocker.MagicMock(event_consumer.GEventMessage)
+    event_msg.msg_id = pubsub_message.message_id
+    event_msg.data = event_msg_data
+    event_msg.phase = ''
+    return event_msg
+
+
+@pytest.fixture
+def mock_http_client(mocker, get_mock_coro):
+    get_json_mock, get_json_coro = get_mock_coro()
+    request_post_mock, post_request_coro = get_mock_coro()
+
+    http_client = mocker.MagicMock()
+
+    mocker.patch.object(http_client, 'get_json', get_json_coro)
+    mocker.patch.object(http_client, '_get_json_mock', get_json_mock)
+
+    mocker.patch.object(http_client, 'request', post_request_coro)
+    mocker.patch.object(http_client, '_request_post_mock', request_post_mock)
+
+    return http_client
+
+
+@pytest.fixture
+def publisher_instance(mock_http_client, config):
+    success, error = asyncio.Queue(), asyncio.Queue()
+    pb = publisher.GDNSPublisher(config, success, error, mock_http_client)
+    return pb
+
+
```

Comment:
The fact that there are 230 lines of fixtures is a little bit of a code smell to me. The more the test data is abstracted away from the tests, the harder it is to tell if you're actually testing the thing you mean to be testing. Other people might have different opinions (@econchick feel free to chime in), but I would use the rule of thumb that an object should be used by at least two tests before it needs to be abstracted into a fixture.